src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -172,4 +179,44 @@ private void validateColumnForPartition(@Nullable String columnName, @Nullable S
                                                     columnName, type));
     }
   }
+
+  private void validateClusteringOrder(@Nullable Schema schema) {
+    if (!shouldCreatePartitionedTable() || Strings.isNullOrEmpty(clusteringOrder) || schema == null) {
+      return;
+    }
+    List<String> columnsNames = Arrays.stream(clusteringOrder.split(",")).map(String::trim)
+      .collect(Collectors.toList());
+    if (columnsNames.size() > MAX_NUMBER_OF_COLUMNS) {
+      throw new InvalidConfigPropertyException(
+        String.format("Expected no more than '%d' clustering fields, found '%d'.", MAX_NUMBER_OF_COLUMNS,
+                      columnsNames.size()), NAME_CLUSTERING_ORDER);
+    }
+    for (String column : columnsNames) {
+      Schema.Field field = schema.getField(column);
+      if (field == null) {
+        throw new InvalidConfigPropertyException(
+          String.format("Clustering column '%s' is missing from the table schema", column),
+          NAME_CLUSTERING_ORDER);
+      }
+      Schema.Type type  = field.getSchema().isNullable()
+        ? field.getSchema().getNonNullable().getType()
+        : field.getSchema().getType();
+      Schema.LogicalType logicalType = field.getSchema().isNullable()
+        ? field.getSchema().getNonNullable().getLogicalType()
+        : field.getSchema().getLogicalType();
+      if (!BigQueryUtil.SUPPORTED_CLUSTERING_TYPES.contains(type)) {
+        throw new InvalidConfigPropertyException(
+          String.format("Field '%s' has type '%s', which is not supported for clustering.", column, type),
+          NAME_CLUSTERING_ORDER);
+      }
+      if ((Schema.Type.INT.equals(type) && !Schema.LogicalType.DATE.equals(logicalType))
```

Comment:
don't need to check the type here, just the logical type.

Reply:
For example: 
we have types that didn't contained Logical part - Schema.Type.STRING, Schema.Type.BOOLEAN. For this type is not necessary check LogicalType.
Other case Schema.Type.BYTES with Schema.LogicalType.DECIMAL that related to BQ type NUMBER. In this case need check Schema.Type.BYTES has Schema.LogicalType.DECIMAL and this check specific only for Schema.Type.BYTES.
If Schema.Type check will be remove system can't correct validate avalible types for clustering fields.

Reply:
What I meant is that this if condition does not need to check the physical type, it only needs to check the logical type. A logical type always has the same physical type. This:

```
Schema.Type.INT == type && Schema.LogicalType.DATE == logicalType
```

is logically equivalent to:

```
Schema.LogicalType.DATE == logicalType
```